Origin: https://marc.info/?l=linux-xfs&m=174031994931608
From: Bastian Germann <bage@debian.org>
Date: Sun, 23 Feb 2025 15:10:52 +0100
Subject: mkfs: Correct filesize declaration

The filesize() implementation was changed to return off_t.
Its declaration still has long. Fix the declaration.

Fixes: 73fb78e5ee8940 ("mkfs: support copying in large or sparse files")
Signed-off-by: Bastian Germann <bage@debian.org>
---
diff --git a/mkfs/proto.c b/mkfs/proto.c
index 6dd3a200..981f5b11 100644
--- a/mkfs/proto.c
+++ b/mkfs/proto.c
@@ -20,7 +20,7 @@ static struct xfs_trans * getres(struct xfs_mount *mp, uint blocks);
 static void rsvfile(xfs_mount_t *mp, xfs_inode_t *ip, long long len);
 static int newregfile(char **pp, char **fname);
 static void rtinit(xfs_mount_t *mp);
-static long filesize(int fd);
+static off_t filesize(int fd);
 static int slashes_are_spaces;
 
 /*
